packages/lesswrong/components/comments/CommentsViews.tsx

```diff
@@ -10,19 +10,35 @@ import type { Option } from '../common/InlineSelect';
 import { getCommentViewOptions } from '../../lib/commentViewOptions';
 import { useNavigate } from '../../lib/reactRouterWrapper';
 
-const CommentsViews = ({post}: {post?: PostsDetails}) => {
+const CommentsViews = ({post, setRestoreScrollPos}: {post?: PostsDetails, setRestoreScrollPos?: (pos: number) => void}) => {
   const currentUser = useCurrentUser();
   const navigate = useNavigate();
   const location = useLocation();
   const { query } = location;
 
   const {InlineSelect} = Components
 
+  // permalinkedCommentHeight() finds the height of the comment at the top of comment permalink pages
+  // on the EA Forum and LessWrong. This is used when the user changes the comment sort order, which
+  // changes the page from a comment permalink page to a regular post page, and we want to preserve
+  // the user's (apparent) scroll position. With that top comment removed, we need to scroll up by its
+  // height.
```

Comment:
> which changes the page from a comment permalink page to a regular post page

The behavior handleViewClick seems quite tightly coupled the the contingent behavior of the navigation. This seems fragile and if it breaks we won't notice. My recommendation: calculate the diff between the new query and the old query, and if we've removed `?commentId=` ... ugh, I don't like that very much either.